iRecruit-ui/src/components/Home/Home.jsx

```diff
@@ -79,37 +80,48 @@ export default function Home({
           placeholder="Search jobs"
         />
       </div>
-      <div className="posts-container">
-        {filteredPosts.length === 0 ? (
-          <p>No jobs found matching the search criteria.</p>
-        ) : (
-          filteredPosts.map((job) => (
-            <div className="post" key={job.slug}>
-              <h2>{job.title}</h2>
-              <h3>{job.company}</h3>
-              <p>Location: {job.location}</p>
-              <p>Date Added: {job.dateAdded}</p>
-              {selectedPostId === job.slug ? (
-                <>
-                  <button onClick={handleCloseApplication}>
-                    Close Application
+
+      {loading ? (
+        <div className="loading-spinner">
+          <TbLoaderQuarter
+            className="spinner-icon"
+            size={45}
+            color="blue"
+            />
+        </div>
+      ) : (
+        <div className="posts-container">
+          {filteredPosts.length === 0 ? (
+            <p>No jobs found matching the search criteria.</p>
+          ) : (
+            filteredPosts.map((job) => (
+              <div className="post" key={job.slug}>
+                <h2>{job.title}</h2>
+                <h3>{job.company}</h3>
+                <p>Location: {job.location}</p>
+                <p>Date Added: {job.dateAdded}</p>
+                {selectedPostId === job.slug ? (
+                  <>
+                    <button onClick={handleCloseApplication}>
+                      Close Application
+                    </button>
+                    <Room
+                      jobId={selectedPostId}
+                      onApplicationSubmit={handleApplicationSubmit}
+                      title={job.title}
+                      joburl={job.url}
+                    />
+                  </>
```

Comment:
This should probably be a separate component or a function that generates this

Reply:
Thank you. Please give me more context on what you mean.